vavr/src/main/java/org/jdbi/v3/vavr/VavrTupleRowMapperFactory.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.vavr;
+
+import io.vavr.CheckedFunction1;
+import io.vavr.Tuple;
+import io.vavr.Tuple0;
+import io.vavr.Tuple1;
+import io.vavr.Tuple2;
+import io.vavr.Tuple3;
+import io.vavr.Tuple4;
+import io.vavr.Tuple5;
+import io.vavr.Tuple6;
+import io.vavr.Tuple7;
+import io.vavr.Tuple8;
+import io.vavr.collection.Array;
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.mapper.ColumnMappers;
+import org.jdbi.v3.core.mapper.NoSuchMapperException;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.mapper.RowMapperFactory;
+import org.jdbi.v3.core.mapper.SingleColumnMapper;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.sql.SQLException;
+import java.util.Optional;
+
+import static org.jdbi.v3.core.generic.GenericTypes.getErasedType;
+import static org.jdbi.v3.core.generic.GenericTypes.resolveType;
+
+public class VavrTupleRowMapperFactory implements RowMapperFactory {
+
+    @Override
+    public Optional<RowMapper<?>> build(Type type, ConfigRegistry config) {
+        Class<?> erasedType = getErasedType(type);
+        if (Tuple0.class.equals(erasedType) || Tuple.class.equals(erasedType)) {
+            return Optional.of((rs, ctx) -> Tuple.empty());
```

Comment:
valid point. actually I was considering throwing an exception in the case of Tuple0 since nothing will be mapped - which is most likely an error on the client side (typo, import of the wrong class). should i throw? i cannot think of any use case where you would query for tuple0

Reply:
btw: you are right, raw `Tuple` is unreasonable, especially since we dont know the expected arity

Reply:
I decided to leave both Tuple and Tuple0 unmapped. Essentially I think it should behave excatly like a call of `build(Void.class, ...)` because you essentially request an inherently empty result type

Reply:
Don't throw an exception--just return `Optional.empty()` to say that the factory doesn't map it. And be explicit in the docs that `Tuple0` and raw `Tuple` are unmapped.